### PR TITLE
MON-3215: drop techpreview gate for collection profiles

### PR DESCRIPTION
**Pull request by rexagod**
* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: pkg/manifests/config.go [hunk 1]
```diff
@@ -52,7 +52,6 @@ const (
 type Config struct {
 	Images      *Images `json:"-"`
 	RemoteWrite bool    `json:"-"`
-	TechPreview bool    `json:"-"`
```
Comment (slashpai) [inline pkg/manifests/config.go:55]: We have [metrics-server PR](https://github.com/openshift/cluster-monitoring-operator/pull/2022) also using same config so I think we can still keep this config and drop the check only for collection profile?

File: pkg/manifests/config.go [hunk 2]
```diff
@@ -190,7 +189,7 @@ func (cps CollectionProfiles) String() string {
 	return sb.String()
 }
 
-func NewConfig(content io.Reader, tp bool) (*Config, error) {
+func NewConfig(content io.Reader) (*Config, error) {
```
Comment (slashpai) [inline pkg/manifests/config.go:192]: same comment above regarding metrics-server

File: pkg/manifests/config.go [hunk 3]
```diff
@@ -458,14 +449,14 @@ func calculateBodySizeLimit(podCapacity int) string {
 // NewConfigFromString transforms a string containing configuration in the
 // openshift-monitoring/cluster-monitoring-configuration format into a data
 // structure that facilitates programmatical checks of that configuration. The
-// content of the data structure might change if TechPreview is enabled (tp), as
+// content of the data structure might change if TechPreview is enabled, as
 // some features are only meant for TechPreview.
-func NewConfigFromString(content string, tp bool) (*Config, error) {
```
Comment (slashpai) [inline pkg/manifests/config.go:463]: same comment above regarding metrics-server

**Comment by simonpasquier** [comment]
I agree with @slashpai to keep the TP bool because it will be reusable in the future.

**Comment by rexagod** [comment]
- [ ] Replace TP with FG.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # Note: This CHANGELOG is only for the monitoring team to track all monitoring related changes. Please see OpenShift release notes for official changes.
 
+## 4.16
+
+- [#2046](https://github.com/openshift/cluster-monitoring-operator/pull/2046) Drop TechPreview gate for CollectionProfiles.
```
Comment (slashpai) [inline CHANGELOG.md:5]: IIUC we added this as a TechPreviewNoUpgrade FeatureGate so for this PR  there is no user change log?

Comment (rexagod) [reply CHANGELOG.md:5]: ACK, the TP was replaced by the newly introduced ways of using feature-gates in https://github.com/openshift/cluster-monitoring-operator/pull/2153. At the moment there is no changelog entry for this, but we might want to point out to users that this is available in TP behind a certain FG if they want to use it. WDYT?

**Pull request closed without merging**